warehouse/manage/views.py

```diff
@@ -616,6 +616,70 @@ def delete_webauthn(self):
         return HTTPSeeOther(self.request.route_path("manage.account"))
 
 
+@view_defaults(
+    uses_session=True,
+    require_csrf=True,
+    require_methods=False,
+    permission="manage:user",
+    http_cache=0,
+    has_translations=True,
+)
+class ProvisionRecoveryCodesViews:
+    def __init__(self, request):
+        self.request = request
+        self.user_service = request.find_service(IUserService, context=None)
+
+    @view_config(
+        request_method="GET",
+        route_name="manage.account.recovery-codes.generate",
+        renderer="manage/account/recovery_codes-provision.html",
+    )
+    def recovery_codes_generate(self):
```

Comment:
Here and for regenerate: does it make sense to allow users to generate recovery codes if they don't have an active second factor method? The template hides this logic in the WebUI if the user doesn't have any 2FA, but it might also make sense to enforce it in the view itself.